docs: Vital Onboarding doc review and cleanup

misterjones opened the pull request:
## Changes

Performed doc review and clean-up on the following pages:

- "Reaching Inside"
- "Anatomy of a Token"

Additional Items:

- Added "Getting Started" section to Table of Contents.
- Added "Extending and Composing Tokens" page to Table of Contents.

🚧 [WIP]: More is coming.

misterjones committed:
Add "Getting Started" section to Table of Contents

misterjones committed:
Update markdown to 100 column-width

misterjones committed:
Doc-cleanup: "Reaching Inside" page

- Minor tweaks to punctuation, grammar, typos, etc.

misterjones committed:
Update "Anatomy of a Token" to 100 column-width

misterjones committed:
Update VDS Table of Contents

- Add "Extending and Composing Tokens" page.
- Fix spacing issue with "Anatomy of a Token" page in the sidebar.

misterjones committed:
Update broken link

- "Next" link to "Anatomy of a Token" page had to be updated due to
  change in ToC value of the page (`AnatomyOfA_Token`).

misterjones committed:
lower-case "higher-order components" (consistency)

misterjones committed:
Doc-cleanup: "Anatomy of a Token" page

- Minor tweaks to punctuation, grammar, typos, etc.
- Add missing links.

misterjones committed:
Complete "Composing from Outside" rename

- Update H1 to match filename.
- Update Table of Contents.
- Update links to page.

misterjones committed:
Update "Working with Content" to 100 column-width

misterjones committed:
Doc-cleanup: "Working with Content" page

- Minor tweaks to punctuation, grammar, typos, etc.
- Add missing links.

misterjones committed:
Update "Extending and Composing" to 100 col-width

wodenx commented on packages/vital-elements/bodiless.docs.json:
```diff
@@ -1,11 +1,25 @@
+      "AnatomyOfA_Token.md": "./doc/GettingStarted/AnatomyOfAToken.md",
+      "FunctionalClasses.md": "./doc/GettingStarted/FunctionalClasses.md",
+      "WorkingWithContent.md": "./doc/GettingStarted/WorkingWithContent.md",
+      "CleanComponents.md": "./doc/GettingStarted/CleanComponents.md",
```
I think these last 2 can be deleted (CleanComponents and YourFirstToken)

misterjones replied:
✅ Deleted.

wodenx commented on packages/vital-elements/doc/ExtendingAndComposing.md:
```diff
@@ -1,15 +1,15 @@
 
-In the examples below, we'll use [the simple `Dialog` component from the 
-`Getting Started` guide](() as our base componnent.
+In the examples below, we'll use [the simple `Dialog` component from the `Getting Started` guide](()
```
Can you fix this link?

misterjones replied:
✅ Fixed — along with others.

wodenx commented on packages/vital-elements/doc/ExtendingAndComposing.md:
```diff
@@ -1,15 +1,15 @@
-for the style and behavior of a component, but in almost every case you will
-extend or compose them to meet hte specific requirements of your site,
-reusing what fits your use case, and overriding or extending what does not.
+Vital tokens are designed to be customized.  They provide a starting point for the style and
```
Hm - I think we may have diffferent settings on our markdown formatters :).  We should probably agree on a standard.  Here's what I'm using.  Please adopt unless you feel strongly about it:
![image](https://github.com/wodenx/Bodiless-JS/assets/480145/4e81ddcf-4e66-4019-857e-d11cb7ace3cc)
And we should add the plugin suggestion and the default settings tot her epo.

wodenx replied:
tbh i don't care what standard we adopt - so if you prefer whatever method you are using, that's fine - but let's get it into the repo so that everyone is using it.

misterjones replied:
I adopted 100 column-width partially because the ruler is set to 100 in the settings ([`/.vscode/settings.json`](https://github.com/johnsonandjohnson/Bodiless-JS/blob/main/.vscode/settings.json#L34-L36)), and partially because 80 felt too constricting anyway.

- I'd prefer 100 column-width, if you're ok with it.
- I've been using the [Rewrap](https://marketplace.visualstudio.com/items?itemName=stkb.rewrap) extension, but I can switch to using [Reflow Markdown](https://marketplace.visualstudio.com/items?itemName=marvhen.reflow-markdown), as you are — I don't think I have a preference there.

Partially related — I've been using these extensions for Markdown, as well, if you think they should be added to the [extension recommendations](https://github.com/johnsonandjohnson/Bodiless-JS/blob/main/.vscode/extensions.json):
- [Code Spell Checker](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker)
- [markdownlint](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint)

misterjones committed:
Remove unnecessary pages

- Delete:
  - "Clean Components"
  - "Your First Token"
- Remove deleted pages from Table of Contents.

misterjones committed:
Doc-cleanup: "Extending and Composing Tokens" page

- Minor tweaks to punctuation, grammar, typos, etc.
- Provide missing links.

misterjones commented on packages/vital-elements/doc/ExtendingAndComposing.md:
```diff
@@ -288,44 +294,46 @@ const WithRedTitle = asDialogToken({
-intended to be composed with another token to produce a variation, a "Variator".
+This pattern is useful when your component has many dimensions of variation. You can see it in full
+force with
+[`vitalCard`](https://github.com/johnsonandjohnson/Bodiless-JS/blob/main/packages/vital-card/src/components/Card/tokens/vitalCard.ts).
```
@wodenx Was this the intended `vitalCard` file?

misterjones commented on packages/vital-elements/doc/ExtendingAndComposing.md:
```diff
@@ -340,95 +348,97 @@ const WithHorizontalOrientation = asDialogToken({
-Variators are usually applied "outside" the tokens they extend, so they 
+The composition pattern is great when you want to add some styling or functionality to an existing
+token, but it's not so effective when you want to _remove_ something. Variators are usually applied
+"outside" the tokens they extend, so they 
```
@wodenx This sentence is incomplete. I wasn't able to guess how you intended to finish it, so I left it as is.
